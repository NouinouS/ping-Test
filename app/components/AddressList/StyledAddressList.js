import styled from 'styled-components';
import List from 'components/List';
import Ul from 'components/List/Ul';

const StyledList = styled(List)`
  grid-row: 5/-1;
  grid-column: 6/8;
  & > ${Ul} {
    { 
      padding-top: 5px;
      max-height: 100%;
    }
`;

export default StyledList;