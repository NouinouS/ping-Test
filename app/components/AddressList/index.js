/**
 *
 * AddressList
 *
 */

import React, { memo } from 'react';
import StyledAddressList from './StyledAddressList';
import List from 'components/List';
import ListItem from 'components/ListItem';
import LoadingIndicator from 'components/LoadingIndicator';
import AddrsListItem from '../../containers/AddrsListItem';

function AddressList({ loading, error, addrs }) {

  if (loading) {
    return <List component={LoadingIndicator} />;
  }

  if (error !== false) {
    const ErrorComponent = () => (
      <ListItem item="Something went wrong, please try again!" />
    );
    return <List component={ErrorComponent} />;
  }

  if (addrs !== false) {
    return <StyledAddressList items={addrs} component={AddrsListItem} />;
  }

  return null;
}

AddressList.propTypes = {};

export default memo(AddressList);
