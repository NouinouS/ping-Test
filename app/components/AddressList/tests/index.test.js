/**
 *
 * Tests for AddressList
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AddressList from '../index';
// import { DEFAULT_LOCALE } from '../../../i18n';

describe('<AddressList />', () => {

  let shallowExpect; 

  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
    shallowExpect = compose(expect,toJSON,shallow);
  });

  it('should render the loading indicator when its loading', () => {
    shallowExpect(<AddressList loading />).toMatchSnapshot();
  });

  it('should render an error if loading failed', () => {
    const { queryByText } = render(
      <IntlProvider locale="en">
        <AddressList loading={false} error={{ message: 'Loading failed!' }} />
      </IntlProvider>,
    );
    expect(queryByText(/Something went wrong/)).not.toBeNull();
  });

  it('Should render and match the snapshot', () => {
    shallowExpect(<AddressList error={false} />).toMatchSnapshot();
  });
});
