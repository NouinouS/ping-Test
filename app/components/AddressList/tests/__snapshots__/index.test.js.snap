// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressList /> Should render and match the snapshot 1`] = `
<StyledAddressList__StyledList
  component={
    Object {
      "$$typeof": Symbol(react.memo),
      "WrappedComponent": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": [Function],
      },
      "compare": null,
      "displayName": "Connect(Component)",
      "type": [Function],
    }
  }
/>
`;

exports[`<AddressList /> should render the loading indicator when its loading 1`] = `
<List
  component={[Function]}
/>
`;
